xonsh/history/sqlite.py

```diff
@@ -67,6 +67,13 @@ def _xh_sqlite_create_history_table(cursor):
         except sqlite3.OperationalError:
             pass
 
+        # add index on inp. since we query when erasedups is True
+        cursor.execute(
+            f"""\
+CREATE INDEX IF NOT EXISTS  idx_inp_history
+ON {XH_SQLITE_TABLE_NAME}(inp);"""
+        )
+
```

Comment:
Maybe a few commits that belonged to a different PR got pulled in here?

Reply:
Yes probably the news items 1&2 could go into a separate PR. but due to some laziness :)  hope you don't mind it. the commits are contained though with proper message.